lib/rustler_precompiled.ex

```diff
@@ -238,35 +260,67 @@ defmodule RustlerPrecompiled do
         nif_versions: nif_versions,
         version: version
       } ->
-        for target_triple <- targets, nif_version <- nif_versions do
-          target = "nif-#{nif_version}-#{target_triple}"
+        all_tar_gzs =
+          for target_triple <- targets, nif_version <- nif_versions do
+            target = "nif-#{nif_version}-#{target_triple}"
 
-          # We need to build again the name because each arch is different.
-          lib_name = "#{lib_prefix(target)}#{basename}-v#{version}-#{target}"
+            # We need to build again the name because each arch is different.
+            lib_name = "#{lib_prefix(target)}#{basename}-v#{version}-#{target}"
 
-          tar_gz_file_url(base_url, lib_name_with_ext(target, lib_name))
-        end
+            [
+              tar_gz_file_url(base_url, lib_name_with_ext(target_triple, lib_name))
+              | maybe_variants_tar_gz_urls(metadata[:variants], base_url, target_triple, lib_name)
+            ]
+          end
 
-      _ ->
-        raise "metadata about current target for the module #{inspect(nif_module)} is not available. " <>
-                "Please compile the project again with: `mix compile --force`"
+        {:ok, List.flatten(all_tar_gzs)}
+
+      wrong_meta ->
+        {:error, wrong_meta}
+    end
+  end
+
+  defp maybe_variants_tar_gz_urls(nil, _, _, _), do: []
+
+  defp maybe_variants_tar_gz_urls(variants, base_url, target_triple, lib_name)
+       when is_map_key(variants, target_triple) do
+    variants = Map.fetch!(variants, target_triple)
+
+    for variant <- variants do
+      tar_gz_file_url(
+        base_url,
+        lib_name_with_ext(target_triple, lib_name <> "--" <> Atom.to_string(variant))
+      )
     end
   end
 
+  defp maybe_variants_tar_gz_urls(_, _, _, _), do: []
+
   @doc """
-  Returns the file URL to be downloaded for current target.
+  Returns the file URLs to be downloaded for current target.
 
+  It is in the plural because a target may have some variants for it.
   It receives the NIF module.
   """
-  def current_target_nif_url(nif_module) when is_atom(nif_module) do
+  def current_target_nif_urls(nif_module) when is_atom(nif_module) do
     metadata =
       nif_module
       |> metadata_file()
       |> read_map_from_file()
 
     case metadata do
       %{base_url: base_url, file_name: file_name} ->
-        tar_gz_file_url(base_url, file_name)
+        target_triple = target_triple_from_nif_target(metadata[:target])
+
+        variants =
+          maybe_variants_tar_gz_urls(
+            metadata[:variants],
+            base_url,
+            target_triple,
+            metadata[:lib_name]
+          )
+
+        [tar_gz_file_url(base_url, file_name) | variants]
```

Comment:
Maybe worth encapsulating inside a `tar_gz_urls` function since it is defined both here and above.

Reply:
Nice! I moved to a private function :)